### ISSUE TITLE
This repo no longer contains any cypress auth implementation

### ISSUE THREAD

**Issue by PossiblyMax** [danlourenco/auth0-cypress-demo#6]
Hi, 

I came to this repo via the link in the auth0 blog post: [End to end testing with cypress and auth0](https://auth0.com/blog/end-to-end-testing-with-cypress-and-auth0/) and found that it is incomplete, in fact the files have been removed that were doing the actual auth implementation.

Even with the original files, they implementation does not work using the new (recommended?) `@auth0/auth0-spa-js` library - implemented as per the instructions on the auth0 post [React: Login](https://auth0.com/docs/quickstart/spa/react/01-login), which uses `code=` on the URL rather than the (older?) `access-token=` that was in the old implementation files of this repository.

Could we get this repo and the blog post updated to be current or with some sort of disclaimer that we can no longer use this method for auth with Cypress?

Thanks!
Max

**Comment by JoeEarly** [comment]
Same as you bud, landed here from that link to find some bellend has removed the code for the example. 

I am in a TS envir so yous may be different but I got it going with the below: 

`Cypress.Commands.add("login", (overrides = {}) => {
  Cypress.log({
    name: "loginViaAuth0"
  });

  const options: Cypress.RequestOptions = {
    method: "POST",
    url: Cypress.env("auth_url"),
    log: true,
    auth: {},
    body: {
      grant_type: "password",
      username: Cypress.env("auth_username"),
      password: Cypress.env("auth_password"),
      audience: Cypress.env("auth_audience"),
      scope: "openid profile email",
      client_id: Cypress.env("auth_client_id"),
      client_secret: Cypress.env("auth_client_secret")
    },
    failOnStatusCode: false,
    followRedirect: true,
    form: false,
    gzip: true,
    headers: {},
    qs: {},
    timeout: 5000
  };
  cy.request(options);
});`

**Comment by PossiblyMax** [comment]
Cheers Joe, and yes pretty frustrating! We did a very similar thing to you and got to the point where we could get the token back from the auth0 api, but since we're using the auth0-spa.js which requires a code to be passed in (instead of the token), we got blocked again.

I am really hoping we don't end up having to do some hacky 'pretend to be a browser' flow, parsing redirect URLs and whatnot. Been down that route in the past, don't fancy doing it again.

I think we'll have to get in touch with our auth0 account manager in order to progress...

**Comment by dillonco** [comment]
@maxcalvin Did you ever figure it out? I'm having the same problem, we're able to call Auth0, get the access/id token back, but whenever we try to checkSession or parseHash it always returns an error.

I also just realized that there's a fork of this on the official Auth0 blog github account, [link here](https://github.com/auth0-blog/auth0-cypress-demo), but they forgot to update the link [here](https://auth0.com/blog/end-to-end-testing-with-cypress-and-auth0)

**Comment by dwilhel1** [comment]
@maxcalvin I've run into the same exact issue with the ` auth0-spa.js` library. My current company has the "paid support ticket feature" with auth0 and I've posed this question to them. I'll post here if they have any suggestions or further thoughts.

**Comment by thisissami** [comment]
@maxcalvin the files are in a different branch (checkout `master` to find them). but as you said, it doesn't really matter when this whole approach is outdated. :) 

@dwilhel1 I'd love an update once you hear back from Auth0. I've sent them a message as well just now, and will update here too if I hear back.

**Comment by d-neri** [comment]
Ran into this problem as well using the SPA SDK. Has anyone heard back from Auth0 on how to handle this?

**Comment by dwilhel1** [comment]
@danielneri before leaving my former position, I heard back from Auth0 support. I don't have the exact message on hand, but they told me Cypress is unsupported and they anxiously look forward to a fix from the community. I believe they were hoping I'd find a way to get this to work.

My "workaround" at the time was a dedicated Cypress test spec file for logging into the Auth0 system. For my other spec files, I had implemented "if `window.Cypress`, skip auth / route guard logic" so that logging in wasn't necessary to access the pages.

**Comment by Narretz** [comment]
Wow, that's pretty shocking. Meanwhile this repo is still online, with no comment from auth0, and the article https://auth0.com/blog/end-to-end-testing-with-cypress-and-auth0/ is also still online as if it was up-to-date. @danlourenco do you have any information about this? Did you just contribute this article / code to auth0 or do you work for them?

**Comment by danlourenco** [comment]
Hi @Narretz -- I'm sorry for missing these messages earlier! I had created the article for Auth0 as I was scratching my own itch a few years ago on a project that used Auth0 + Cypress.  I have to admit, I need to do some catch-up and read up on what has changed with Auth0 and Cypress since I implemented this solution.  I'll respond once that's done, but I don't have any more information for you at the moment.

**Comment by eplatek-safyr** [comment]
@danlourenco [I came accross an issue](https://github.com/auth0/auth0-spa-js/issues/581) after an update to the last version of the library `@auth0/auth0-spa-js` that has changed its behavior since version `1.12.0`. A simpler approach was provided by @adamjmcgrath at this repo: https://github.com/adamjmcgrath/cypress-spa-example

TLDR: use localStorage instead of memory for cache location and just fill the local storage with the received token:

```js
const client_id = 'wLSIP47wM39wKdDmOj6Zb5eSEw3JVhVp';
const client_secret = '';
const audience = 'https://brucke.auth0.com/api/v2/';
const scope = 'openid profile email';
const username = '';
const password = '';

context('Example', () => {
  it('should be logged in', () => {
    cy.clearLocalStorage();

    cy.log('getting tokens');
    cy.request({
      method: 'POST',
      url: 'https://brucke.auth0.com/oauth/token',
      body: {
        grant_type: 'password',
        username,
        password,
        audience,
        scope,
        client_id,
        client_secret,
      },
    }).then(({ body: { access_token, expires_in, id_token, token_type } }) => {

      cy.visit('http://localhost:3000');
      cy.get('#btn').should('be.visible');
      cy.get('#hello').should('not.be.visible');

      cy.window().then((win) => {
        win.localStorage.setItem(
            `@@auth0spajs@@::${client_id}::${audience}::${scope}`,
            JSON.stringify({
              body: {
                client_id,
                access_token,
                id_token,
                scope,
                expires_in,
                token_type,
                decodedToken: { user: JSON.parse(Buffer.from(id_token.split('.')[1], 'base64').toString('ascii')) },
                audience
              },
              expiresAt: Math.floor(Date.now() / 1000) + expires_in
            })
        )
      })
      cy.visit('http://localhost:3000');

      cy.get('#btn').should('not.be.visible');
      cy.get('#hello').should('contain', `Hello ${username}`);
    });
  })
})
```

**Comment by leonardoinstaleap** [comment]
Hello @danlourenco  do you have any new information?

**Comment by ahelord** [comment]
Hello @danlourenco , do you have any new information?
